language/move-lang/src/parser/syntax.rs

```diff
@@ -1305,37 +1306,72 @@ fn parse_optional_type_args(tokens: &mut Lexer<'_>) -> Result<Option<Vec<Type>>,
     }
 }
 
+fn token_to_ability(token: Tok, content: &str) -> Option<Ability_> {
+    match (token, content) {
+        (Tok::Copy, _) => Some(Ability_::Copy),
+        (Tok::IdentifierValue, Ability_::DROP) => Some(Ability_::Drop),
+        (Tok::IdentifierValue, Ability_::STORE) => Some(Ability_::Store),
+        (Tok::IdentifierValue, Ability_::KEY) => Some(Ability_::Key),
+        _ => None,
+    }
+}
+
+// Parse a type ability
+//      Ability =
+//          <Copy>
+//          | "drop"
+//          | "store"
+//          | "key"
+fn parse_ability(tokens: &mut Lexer) -> Result<Ability, Error> {
```

Comment:
Interesting that this works without "Lexer<'_>". I had observed that as well in one of my changes but kept it to match the other changes from #7654. Maybe we can just clean them all up? (Certainly not as part of this PR!)

Reply:
I'll make another PR later to clean this up. Rust lifetime elision tends to work nested as long as you don't rely on the lifetime